align/constGap.go

```diff
@@ -277,14 +277,29 @@ func writeCigar(trace [][]ColType, i_inChecker_max_Previous int, j_inChecker_max
 }
 
 //Step 4
-//inputs: the growing route describing the collection of cigars of the entire alignment except the last cigar, the index of the cigar that is currently being built, the ColType (M=0,I=1,D=2) of the last cigar entry
+//inputs: lengths of original sequences alpha and beta, the growing route describing the collection of cigars of the entire alignment except the last cigar, the index of the cigar that is currently being built, the ColType (M=0,I=1,D=2) of the last cigar entry
 //outputs: the updated final route describing the collection of cigars of the entire alignment including the last cigar, the updated and index of the cigar that is currently being built
-func lastCigar(route []Cigar, routeIdx_current int, Op_end ColType) ([]Cigar, int) {
+func lastCigar(len_alpha int, len_beta int, route []Cigar, routeIdx_current int, Op_end ColType) ([]Cigar, int) {
+	//Calculate the size of final runlength
+	var TotalRunLength, LastRunLength int64
+	TotalRunLength = 0
+	for routeIdx := range route {
+		TotalRunLength += route[routeIdx].RunLength
```

Comment:
I am not sure we want to count insertions, deletions, and matches here.  I am wondering if (depending on whether we use len_beta or len_alpha) we want to either count only insertions+matches, or deletions+matches

Reply:
also make gapSize threshold customizable to test "doNotCalculate"